provider/gce/environ_broker.go

```diff
@@ -133,7 +145,18 @@ func (env *environ) findInstanceSpec(
 // newRawInstance is where the new physical instance is actually
 // provisioned, relative to the provided args and spec. Info for that
 // low-level instance is returned.
-func (env *environ) newRawInstance(args environs.StartInstanceParams, spec *instances.InstanceSpec) (*google.Instance, error) {
+func (env *environ) newRawInstance(args environs.StartInstanceParams, spec *instances.InstanceSpec) (_ *google.Instance, err error) {
+
+	zoneIndependent := true
+	defer func() {
```

Comment:
I realize there are a number of returns here, but this feels a bit like action-at-a-distance that will ultimately be hard to understand why we're getting ZoneIndependent.

maybe its fine. Though I'd still like to keep Tracing information.
Is it possible to use something like errors.Wrap?

Reply:
Fair enough. I'll inline or use helpers where it feels appropriate.